### PR TITLE
Compliance test

### PR DESCRIPTION
**Pull request by yisraelU**
the Pizza spec from Smithy4s has been replicated here using the smithy test traits

### CODE REVIEW

**Review by daddykotex — COMMENTED** [review]
I don't see an issue here, it looks good, good job

File: modules/core/resources/META-INF/smithy/test/pizza.smithy [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        protocol: simpleRestJson
+        uri : "/custom-code/399"
+        method: "GET"
+        body:""
+        params:{
+            "code": 399
+        }
+    }
+])
+@httpResonsesTests([
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/test/pizza.smithy:301]: `httpResonsesTests`? seems like this doesn't get automatically validated 🤔

File: modules/core/resources/META-INF/smithy/test/pizza.smithy [hunk 2]
```diff
@@ -0,0 +1,324 @@
...
+    {
+        id: "CustomCodeOutput"
+        documentation: "respect the httpresponseCode trait",
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/test/pizza.smithy]: text casing issue in `httpresponseCode`

File: modules/core/resources/META-INF/smithy/test/pizza.smithy [hunk 3]
```diff
@@ -0,0 +1,324 @@
...
+        value: "server.error",
+        name: "ERROR_CODE"
+    }
+])
+string UnknownServerErrorCode
+
+
+@trait
+document freeForm
+
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/test/pizza.smithy:249]: is this even necessary for these tests?

File: modules/core/resources/META-INF/smithy/test/pizza.smithy [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+    @required
+    errorCode: UnknownServerErrorCode,
+
+    description: String,
+
+    stateHash: String
+}
+
+// Define the singular error code that can be returned for an UnknownServerError
+@enum([
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/test/pizza.smithy:238]: why not use a 2.x enum?

File: modules/core/resources/META-INF/smithy/test/pizza.smithy (hunk 1 above)
Comment (yisraelU) [reply modules/core/resources/META-INF/smithy/test/pizza.smithy]: apparently not

Comment (yisraelU) [reply modules/core/resources/META-INF/smithy/test/pizza.smithy]: fixed

File: modules/core/resources/META-INF/smithy/test/CustomCode.smithy [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        protocol: simpleRestJson
+        uri : "/custom-code/399"
+        method: "GET"
+        body:""
+        params:{
+            "code": 399
+        }
+    }
+])
+apply CustomCode @httpResonseTests([
```
Comment (kubukoz) [inline modules/core/resources/META-INF/smithy/test/CustomCode.smithy:23]: missed this one :P

**Review by kubukoz — CHANGES_REQUESTED** [review]

File: modules/core/resources/META-INF/smithy/test/CustomCode.smithy (hunk 5 above)
Comment (yisraelU) [reply modules/core/resources/META-INF/smithy/test/CustomCode.smithy:23]: lol , i hate this

File: modules/core/resources/META-INF/smithy/test/pizza.smithy (hunk 4 above)
Comment (yisraelU) [reply modules/core/resources/META-INF/smithy/test/pizza.smithy:238]: i should , i just copied this over from smithy4s

File: modules/core/resources/META-INF/smithy/test/pizza.smithy (hunk 3 above)
Comment (yisraelU) [reply modules/core/resources/META-INF/smithy/test/pizza.smithy:249]: not sure why it was here in the first place. will do a cleanup

**Comment by yisraelU** [comment]
@kubukoz in general i think there is a lot of redundancy amongst these tests . I think we should  probably copy over the relevant aws tests that the use for restJson1 as most of them will apply and are testing for very specific adherence

**Comment by kubukoz** [comment]
any clue why CI didn't find the typos?

File: modules/core/resources/META-INF/smithy/test/AddMenuItem.smithy [hunk 6]
```diff
@@ -0,0 +1,78 @@
+$version: "2"
```
Comment (Baccata) [inline modules/core/resources/META-INF/smithy/test/AddMenuItem.smithy:1]: the tests should be in a separate module altogether, not in the core module.

File: build.sc [hunk 7]
```diff
@@ -148,6 +148,13 @@ class OpenapiModule(crossVersion: String) extends BaseCrossScalaModule {
   object test extends this.Tests with BaseMunitTests
 }
 
+object tests extends BaseScalaModule {
+
+
+  def moduleDeps = Seq(core)
+
+}
```
[the remark this answers is not in the document]
Comment (yisraelU) [reply build.sc:156]: will do

**Review by Baccata — APPROVED** [review]

**Comment by Baccata** [comment]
@yisraelU I've taken the liberty to rename the `tests` module to `protocol-tests` to align with the module from the aws library, and also to make the directory structure a little bit clearer.

**Pull request merged**